src/chrome/content/rules/Adsurve.com.xml

```diff
@@ -3,20 +3,17 @@
 
 		- www01		(refused)
 
-
-	Expired 2013-11-20
-
 -->
-<ruleset name="adsurve.com (partial)" default_off="expired">
+<ruleset name="Adsurve.com (partial)">
 
 	<target host="adsurve.com" />
 	<target host="www.adsurve.com" />
 
 
-	<securecookie host="^adsurve\.com$" name=".+" />
+	<securecookie host="^\w" name=".+" />
```

Comment:
why `"^\w"` instead of `".+"` ?

Reply:
To be quite honest, I don't really understand what I'm doing with `securecookie`s, but as I understand it `^\w` secures cookies specific to a particular host, and `.+` secures domain-wide cookies that might be accessed by HTTP-only subdomains. This could be very wrong though. In this instance I haven't actually done any cookie testing and therefore I've tried to match the existing behaviour as closely as possible.

Reply:
As I understand it, `.+` secures cookies for all targets and `^\w` secures cookies for all targets with only alphabetic characters.

So they should be functionally equivalent (as long as you have have digits in your targets) but `^\w` won't be read as a trivial securecookie rule by the rewriter.

Only subdomains listed as targets can be affected by securecookie and rewrite rules.

But that's just my take on it, feel free to wait for another opinion.

Reply:
@bisaloo Only the first character should be alphabetic.

Reply:
In this case, better use `.+`.

Reply:
@Bisaloo [`^\w` matches any alpha**numeric** character including the underscore](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#special-word). This means that all cookies will be accepted, except those starting with dots – however it appears that my understanding of the difference between `example.com`and `.example.com` [no longer applies](https://stackoverflow.com/a/20884869/1414639). Perhaps @J0WI can clarify.

I've made the change to this ruleset though.

Reply:
About the leading dot, see https://www.eff.org/https-everywhere/rulesets#secure-cookies

> Non-HTTPS subdomains that are not listed as targets may depend on cookies set on the root domain, that we secure and stop them from accessing.

True but that doesn't change the fact that `^\w` and `.+` should be functionally equivalent.